plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -610,6 +610,10 @@ def update_column_descriptions(self, relation, columns):
                 column_config = columns[column.name]
                 column_dict = column.to_api_repr()
                 column_dict['description'] = column_config.get('description')
+                if column_config.get('tags'):
```

Comment:
Let's not use `tags` directly -- can we instead create a new config called `policy_tags` and use it here? That might involve reaching into the `Column` contracts defined in dbt.

Alternatively, I do like the idea of repurposing a property in the `meta` dict for this config. I know @jtcohen6 voiced his opposition because meta is intended to be purely informational, which is correct, but I'm looking through the code now and I think adding a new column-level property specific to BigQuery might end up being a can of worms!

@jtcohen6 @beckjake - what do you two think?

Reply:
I do agree with @jtcohen6 that `meta` should stay informational/user-controlled.

I think we are going to have to deal with the idea of adapter-specific column fields (and adapter-specific schema.yml entries), much like we did with `config`. Maybe as a first pass, we could change `UnparsedColumn` (or even its ancestor class `HasDocs`) to derive from `ExtensibleJsonSchemaMixin` so it can get arbitrary keys? I haven't really thought about the consequences of that at all, but it's what I'd look at first.

Reply:
I'm supportive of core's classes (`HasDocs`, `ColumnInfo`) accepting arbitrary keys, as long as there's some way for plugin authors to enumerate those additional keys, similar to `AdapterSpecificConfigs`.

Reply:
If having `HasDocs` derive from `ExtensibleJsonSchemaMixin` is the correct solution here, any guidance on how to setup an `AdapterSpecificConfigs` equivalent for this case?

Reply:
Hey @jtcohen6 and @beckjake, any thoughts on the above question?

Reply:
@azhard sorry about this! For example, here's what I have going in a branch currently to solve a similar problem:

```
class AdditionalPropertiesMixin:
    """Make this class an extensible property.

    The underlying class definition must include a type definition for a field
    named '_extra' that is of type `Dict[str, Any]`.
    """
    @classmethod
    def from_dict(cls, data, validate=True):
        self = super().from_dict(data=data, validate=validate)
        keys = self.to_dict(validate=False, omit_none=False)
        for key, value in data.items():
            if key not in keys:
                self.extra[key] = value
        return self

    def to_dict(self, omit_none=True, validate=False):
        data = super().to_dict(omit_none=omit_none, validate=validate)
        data.update(self.extra)
        return data

    def replace(self, **kwargs):
        dct = self.to_dict(omit_none=False, validate=False)
        dct.update(kwargs)
        return self.from_dict(dct)

    @property
    def extra(self):
        return self._extra

class MyOtherClass(AdditionalPropertiesMixin, ExtensibleJsonSchemaMixin):  # inheritance order matters
    method: str
    value: Union[str, int]
    _extra: Dict[str, Any] = field(default_factory=dict)
```

Reply:
Hey @beckjake, I'm more unsure about how to setup the equivalent of a `AdapterSpecificColumnInfo` so that I can pass use a `BigQueryColumnInfo` with `policy_tags` property here but elsewhere it defaults to `ColumnInfo`.

I know inside `BigQueryAdapter` we set `Column = BigQueryColumn`, I'm assuming we'd want to do something similar for the ColumnInfo but not sure how to ensure that the passed in ColumnInfo gets used elsewhere

Reply:
I think you just want to make `ColumnInfo` itself allow additional properties, and then the bigquery code that wants it can extract them out if it needs them. Making that aspect of parsing itself adapter-specific would be quite an overhaul.

Reply:
Thanks for clarifying @beckjake. Out of curiosity, what's the thinking behind creating the new `AdditionalPropertiesMixin` class vs. using `AdditionalPropertiesAllowed` and removing the default value for `_extra` to get around the `TypeError: non-default argument ... follows default argument` error?

Reply:
Well, this answer is kind of circular reasoning, but: Because I want to have a default value for `_extra`! It's convenient.

Reply:
Gotcha, okay updating accordingly!